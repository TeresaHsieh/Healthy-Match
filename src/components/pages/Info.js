import React from "react";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";
import "../../css/info.css";
import ChefMatch from "../../imgs/chef-match.png";
import Edit from "../../imgs/edit.png";
import BasicInfo from "../../components/others/ComOfInfo/BasicInfo";
import AuthInfo from "../../components/others/ComOfInfo/AuthInfo";
import ContributionInfo from "../others/ComOfInfo/ContributionInfo";
import { checkUserInfo } from "../../store/actions/authAction";
import SignUpRedirect from "../others/ComOfMember/SignUpRedirect";

import LoveMatch from "../../imgs/love-match.png";

// App Components
import Header from "../common/Header";
import { relativeTimeRounding } from "moment";

class Info extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      openFirstSignUpBox: false
    };
  }

  componentDidMount = () => {
    // this.props.checkUserInfo(this.props.auth.uid);
    if (
      // !this.props.userInfo.Name &&
      !this.props.userInfo.Age &&
      !this.props.userInfo.Weight &&
      !this.props.userInfo.Height &&
      !this.props.userInfo.Sexual
    ) {
      this.setState({ openFirstSignUpBox: true });
    }
  };

  handleInfoChange = () => {
    console.log("調整資料");
  };

  closeFirstSignUpBox = () => {
    this.setState({ openFirstSignUpBox: false });
  };

  render() {
    const { auth } = this.props;
    if (!auth.uid) return <Redirect to="./member" />;

    const openFirstSignUpBox = this.state.openFirstSignUpBox;

    return (
      <div>
        <Header />
        <div className="main-info">
          {openFirstSignUpBox ? (
            <div className="backgroundSignUpRedirectBox">
              <div className="signUpRedirectBox">
                <SignUpRedirect />{" "}
                <button onClick={this.closeFirstSignUpBox}>我知道了！</button>
              </div>
            </div>
          ) : (
            ""
          )}
          <div className="simple-info">
            {/* <h1>
              HELLO :)
              <br /> {this.props.userInfo.Name}
            </h1> */}
            {/* <AuthInfo /> */}
          </div>
          <div className="more-info">
            <BasicInfo />
            <ContributionInfo />
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    date: new Date().toLocaleDateString(),
    auth: state.firebase.auth,
    userInfo: state.firebase.profile
  };
};

const mapDispatchToProps = dispatch => {
  return {
    // create a method
    checkUserInfo: userUID => {
      dispatch(checkUserInfo(userUID));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Info);
