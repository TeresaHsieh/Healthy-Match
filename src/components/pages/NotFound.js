import React from "react";
import "../../css/404.css";

// App Components
import Header from "../common/Header";
import SadMatch from "../../imgs/sad-match.png";

const NotFound = () => (
  <div>
    <Header />
    <div className="main-404">
      <div>
        <h2 className="code404"> 404 </h2>
        <h2> 喔不！此頁不存在！ </h2>
        <p> 請再次確認網址路徑～麻吉感謝你 ❤️ </p>
      </div>
      <img src={SadMatch} className="sad-match" />
    </div>
  </div>
);

export default NotFound;
