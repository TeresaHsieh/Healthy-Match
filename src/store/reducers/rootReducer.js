import memberReducer from "./memberReducer";
import dailyReducer from "./dailyReducer";
import { firestoreReducer } from "redux-firestore"; // sync firestore data to our state

import { combineReducers } from "redux";

const rootReducer = combineReducers({
  member: memberReducer,
  daily: dailyReducer,
  firestore: firestoreReducer
});

export default rootReducer;
