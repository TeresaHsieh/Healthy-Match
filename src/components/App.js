import React from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { connect } from "react-redux";
import { Redirect } from "react-router-dom";

// App Components
import Home from "./pages/Home";
import DailyRecord from "./pages/DailyRecord";
import History from "./pages/History";
import MatchStatus from "./pages/MatchStatus";
import Member from "./pages/Member";
import LogOut from "./pages/LogOut";
import News from "./pages/News";
import Info from "./pages/Info";
import NotFound from "./pages/NotFound";

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      loading: true
    };
  }

  componentDidMount() {
    // this simulates an async action, after which the component will render the content
    demoAsyncCall().then(() => this.setState({ loading: false }));
  }

  render() {
    const { loading } = this.state;

    if (loading) {
      // if your component doesn't have to wait for an async action, remove this block
      return null; // render null when app is not ready
    }
    return (
      <BrowserRouter>
        <Switch>
          <Route exact path="/" component={Home} />

          {/* Home */}

          <Route path="/daily-record" component={DailyRecord} />
          <Route path="/history" component={History} />
          <Route path="/match-status" component={MatchStatus} />
          <Route path="/news" component={News} />
          <Route path="/member" component={Member} />
          <Route path="/info" component={Info} />
          <Route path="/log-out" component={LogOut} />

          <Route component={NotFound} />
        </Switch>
      </BrowserRouter>
    );
  }
}

function demoAsyncCall() {
  return new Promise(resolve => setTimeout(() => resolve(), 100));
}

// const mapStateToProps = state => {
//   return {
//     auth: state.firebase.auth
//   };
// };

// export default connect(
//   null,
//   mapStateToProps
// )(App);

export default App;
