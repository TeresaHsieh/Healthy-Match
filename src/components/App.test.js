import React from "react";
import App from "../components/App";
import { cleanup, fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Info from "../components/pages/Info";
import NotFound from "../components/pages/NotFound";
import { createStore, applyMiddleware, compose } from "redux";
import rootReducer from "../store/reducers/rootReducer";
import { Provider } from "react-redux";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { exportAllDeclaration } from "@babel/types";

test("Test Info", () => {
  let store = createStore(rootReducer);
  const notFoundWithRudex = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <NotFound />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(notFoundWithRudex).toMatchSnapshot();

  //   const infoWithRudex = renderer
  //     .create(
  //       <Provider store={store}>
  //         <BrowserRouter>
  //           <Info />
  //         </BrowserRouter>
  //       </Provider>
  //     )
  //     .toJSON();
  //   expect(infoWithRudex).toMatchSnapshot();

  //   const notFoundWithRudex = renderer
  //     .create(
  //       <Provider store={store}>
  //         <BrowserRouter>
  //           <NotFound />
  //         </BrowserRouter>
  //       </Provider>
  //     )
  //     .toJSON();
  //   expect(notFoundWithRudex).toMatchSnapshot();
});
