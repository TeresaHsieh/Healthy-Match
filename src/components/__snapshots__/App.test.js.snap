// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Info 1`] = `
<div>
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="logo"
        src="test-file-stub"
      />
    </a>
    <div>
      <button
        className="hamburger-button"
        onClick={[Function]}
      >
        <div
          className="hamburger-line"
        />
        <div
          className="hamburger-line"
        />
        <div
          className="hamburger-line"
        />
      </button>
      <div />
    </div>
  </header>
  <div
    className="main-404"
  >
    <div>
      <h1
        className="code404"
      >
         404 
      </h1>
      <h2>
         喔不！此頁不存在！ 
      </h2>
      <p>
         請再次確認網址路徑～麻吉感謝你 ❤️ 
      </p>
    </div>
    <img
      className="sad-match"
      src="test-file-stub"
    />
  </div>
</div>
`;

exports[`Test Info 2`] = `
<div>
  <header
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="logo"
        src="test-file-stub"
      />
    </a>
    <div>
      <button
        className="hamburger-button"
        onClick={[Function]}
      >
        <div
          className="hamburger-line"
        />
        <div
          className="hamburger-line"
        />
        <div
          className="hamburger-line"
        />
      </button>
      <div />
    </div>
  </header>
  <div
    className="main-404"
  >
    <div>
      <h1
        className="code404"
      >
         404 
      </h1>
      <h2>
         喔不！此頁不存在！ 
      </h2>
      <p>
         請再次確認網址路徑～麻吉感謝你 ❤️ 
      </p>
    </div>
    <img
      className="sad-match"
      src="test-file-stub"
    />
  </div>
</div>
`;
